maintainers/scripts/update-luarocks-packages.sh

```diff
@@ -0,0 +1,163 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -p 'lua5_2.withPackages(ps: [ ps.luarocks-nix ])' nix-prefetch-scripts -i bash
+
+# lua5_2.withPackages(ps: [ ps.luarocks-nix ])
+# You'll likely want to use
+# ``
+# nixpkgs $ maintainers/scripts/update-luarocks-packages.sh > pkgs/top-level/lua-generated-packages.nix
+# ``
+# to update all libraries in that folder.
+
+# Possible improvements:
+# browse rocks here
+# https://github.com/luarocks/luarocks/wiki/Browse-rocks
+# we might be able to download rocks from
+# http://luafr.org/moonrocks/ or
+# https://github.com/rocks-moonscript-org/moonrocks-mirror.git
+
+# stop the script upon C-C
+set -eu -o pipefail
+
+if [ $# -lt 1 ]; then
+    echo "Usage: $0 GENERATED_FILENAME"
+    echo "(most likely pkgs/top-level/lua-generated-packages.nix)"
+    exit 1
+fi
+
+GENERATED_NIXFILE="$1"
+PACKAGE=""
+# we have to use a temp file else
+TMP_FILE="$(mktemp)"
+
+exit_trap()
+{
+  local lc="$BASH_COMMAND" rc=$?
+  test $rc -eq 0 || echo "*** error $rc: $lc"
+}
+
+trap exit_trap EXIT
+
+
+# -p <PKG> package
+while getopts "p:" opt; do
+  case $opt in
+    p)
+      echo "Generating nix derivation for $OPTARG !" >&2
+      PACKAGE="$OPTARG"
+      ;;
+    \?)
+      echo "Invalid option: -$OPTARG" >&2
+      ;;
+  esac
+done
+
+read -d '' -r HEADER <<EOM
+/* ${GENERATED_NIXFILE} is an auto-generated file -- DO NOT EDIT!
+Regenerate it with:
+nixpkgs$ ${0} ${GENERATED_NIXFILE}
+*/
+{
+# self has buildLuaPackage
+self
+, stdenv
+, fetchurl
+, fetchgit
+, toLuaModule
+, requiredLuaModules
+}:
+with self;
+rec {
+EOM
+
+read -d '' -r FOOTER <<'EOM'
+}
+/* GENERATED */
+EOM
+
+# passing arrays is not easy
+# expects 2 args
+#  - pkgs list
+#  - server or ""
+function convert_pkg_from_list () {
+	local -n pkg_list=$1
+	server=""
+	if [ ! -z "$2" ]; then
+		server=" --server=$2"
+	fi
+
+	for pkg in "${pkg_list[@]}"
+	do
+        convert_pkg "$pkg" "$server"
+	done
+}
+
+function convert_pkg () {
+    pkg="$1"
+	server=""
+	if [ ! -z "$2" ]; then
+		server=" --server=$2"
+	fi
+
+	# shift
+    echo "looking at $pkg from server $server" >&2
+    drv=$(luarocks convert2nix $server "$pkg")
+    if [ $? -gt 0 ]; then
+        echo "Failed to convert $pkg" >&2
+        echo "$drv" >&2
+        # TODO here we could have a flag to pack and upload missing src.rock
+        # to a nixos luarock manifest
+    else
+        echo "$drv" | tee -a "$TMP_FILE"
+    fi
+}
+
+
+declare -A pkg_list=(
```

Comment:
Can you source a file containing `pkg_list` externally to separate program and data.

Reply:
done